engine/battle/effect_commands.asm

```diff
@@ -3248,6 +3249,37 @@ CheckStatHerbs:
 	ldh [hBattleTurn], a
 	ret
 
+CheckThroatSpray:
+	ld a, [wBattleEnded]
+	and a
+	ret nz
+
+	ld a, [wAttackMissed]
+	and a
+	ret nz
+	
+	call HasUserFainted
+	ret z
+
+	predef GetUserItemAfterUnnerve
+	push bc
+	call GetCurItemName
```

Comment:
Move this to above line 3273 and test if everything still works as expected. It may save an extra push/pop.

Reply:
Works!